diff --git a/llex.c b/llex.c
index 88c6790c..68c6d255 100644
--- a/llex.c
+++ b/llex.c
@@ -177,16 +177,7 @@ static void buffreplace (LexState *ls, char from, char to) {
 
 
 static void trydecpoint (LexState *ls, SemInfo *seminfo) {
-  /* format error: try to update decimal point separator */
-  struct lconv *cv = localeconv();
-  char old = ls->decpoint;
-  ls->decpoint = (cv ? cv->decimal_point[0] : '.');
-  buffreplace(ls, old, ls->decpoint);  /* try updated decimal separator */
-  if (!luaO_str2d(luaZ_buffer(ls->buff), &seminfo->r)) {
-    /* format error with correct decimal point: no more options */
-    buffreplace(ls, ls->decpoint, '.');  /* undo change (for error message) */
-    luaX_lexerror(ls, "malformed number", TK_NUMBER);
-  }
+  luaX_lexerror(ls, "malformed number", TK_NUMBER);
 }
 
 
